deploy-0chain/action.yml

```diff
@@ -69,21 +69,13 @@ inputs:
     default: "NONE"
     description: 'custom gosdk version. Will upgrade CLI branches with this before running tests'
 
-
-#          ==================================================
-#          *******!!!!!!!!!!  PLEASE READ   !!!!!!!!!!*******
-#          ==================================================
-#          THE ACTIONS_BRANCH VARIABLE BELOW *MUST* MATCH THE CURRENT BRANCH NAME FOR THE DEPLOY ACTION TO WORK ON THE CURRENT BRANCH
-#          THIS IS A LIMITATION OF HOW GITHUB USES COMPOSITE ACTIONS WHICH WE ARE TRYING TO ADDRESS IN A LESS MANUAL WAY...
-#          ================================
-
 runs:
   using: "composite"
   steps:
     - name: "Setup Config"
       shell: 'script --return --quiet --command "bash {0}"'
       run: |
-        echo "ACTIONS_BRANCH=master" >> $GITHUB_ENV
+        echo "ACTIONS_BRANCH=$(cd ../../_actions/0chain/actions/ && find . -type d  -maxdepth 2 | awk -F/ '{print NF "\t" $0}' | sort -t: -k1 -g -r | head -1 | cut -f2 | cut -c 3-)" >> $GITHUB_ENV
```

Comment:
Yes, I know. lol
This monster of a command derives the current actions branch name by listing the directory used to checkout the action

Reply:
replaced the above command with something less brittle, should be good to merge now